### ISSUE TITLE
You are so awesome!!!!!!!

### ISSUE THREAD

**Issue by bastiaanc** [SharkyKZ/joomla-turnstile-plugin#1]
Thanks so much for this. I was already desperately looking for it!
Love this!!!!

Please see:
https://github.com/BkrBkr/AuthCaptchaForJoomla/issues/4

**Comment by SharkyKZ** [comment]
I don't think I'll be adding workarounds for this. AuthCaptcha plugin essentially doesn't pass any relevant data to captcha plugins. It expects plugins to have a hardcoded field name and read the captcha solution from request, both of which are bad practices. It has some other major issues like initializing captcha plugins on every page, which can lead to bricked sites (try enabling recaptcha plugin before setting up site/secret keys and see what happens). Or using event dispatcher instead of Captcha API to trigger plugins, which leads to failures as well. TL;DR this should be fixed in AuthCaptcha plugin.

**Comment by bastiaanc** [comment]
I understand,

Is there another way to force a captcha in those places where Joomla does not?

**Comment by SharkyKZ** [comment]
No, unless there's an alternative plugin. I submitted a patch https://github.com/BkrBkr/AuthCaptchaForJoomla/pull/5 that should fix the issues mentioned above, in case the plugin is still maintained.

**Comment by bastiaanc** [comment]
> No, unless there's an alternative plugin. I submitted a patch [BkrBkr/AuthCaptchaForJoomla#5](https://github.com/BkrBkr/AuthCaptchaForJoomla/pull/5) that should fix the issues mentioned above, in case the plugin is still maintained.

I made the changes you made, and it seems to work now. It was flawless on the first website. Thank you so much. I am very concerned that the current plugin (AuthCaptcha) maybe will not be updated. I think Joomla! should have this as a standard. Please explain if I am making a thinking mistake and that I might not understand the technique behind it well to make the best decision. Anyway. I am hoping that if "AuthCaptcha" is not maintained that you might be willing to build it better.

Again, thank you so much for helping!

**Comment by SharkyKZ** [comment]
I agree, this is a sensible feature to have out of the box. And it would be trivial to implement. But so far this feature request has been rejected for no clear reason (see https://github.com/joomla/joomla-cms/issues/14791, https://github.com/joomla/joomla-cms/issues/19749, https://github.com/joomla/joomla-cms/issues/35842, https://github.com/joomla/joomla-cms/issues/35866). And because login forms don't use Form API, we're left with having to write these volatile plugins that rely on the generated HTML form output.

**Comment by SharkyKZ** [comment]
https://github.com/BkrBkr/AuthCaptchaForJoomla/pull/5 has been merged so closing this.
